### PR TITLE
Isodose tab reworked to reduce code repetition

### PR DESCRIPTION
**Pull request by Fincap**
No functional differences.

### CODE REVIEW

File: src/View/mainpage/IsodoseTab.py [hunk 1]
```diff
@@ -33,21 +35,22 @@ def init_layout(self):
 
     def init_color_isod(self):
         """
-        Create a dictionary containing the colors for each isodose.
+        Create a list containing the colors for each isodose.
 
-        :return: Dictionary where the key is the percentage of isodose and the value a QColor object.
+        :return: List of QColor objects that correspond to each isodose as defined outside the class..
```
Comment (sjswerdloff) [inline src/View/mainpage/IsodoseTab.py:40]: This is still a Dictionary

File: src/View/mainpage/IsodoseTab.py [hunk 2]
```diff
@@ -67,65 +70,26 @@ def init_checkboxes(self):
...
-        list_of_checkboxes.append(checkbox3)
-        list_of_checkboxes.append(checkbox4)
-        list_of_checkboxes.append(checkbox5)
-        list_of_checkboxes.append(checkbox6)
-        list_of_checkboxes.append(checkbox7)
-        list_of_checkboxes.append(checkbox8)
-        list_of_checkboxes.append(checkbox9)
-        list_of_checkboxes.append(checkbox10)
+        first_iteration = True
+        for i in range(10):
```
Comment (sjswerdloff) [inline src/View/mainpage/IsodoseTab.py:80]: or have one line for i = 0 and then range(1:10) for the rest.

while the following is not ideal for this situation, I've learned about enumerate
```
for i, isodose_percent in enumerate(isodose_percentages):
  print(i, isodose_percent) # or whatever else you need the count/index and value for...
```

**Review by sjswerdloff — APPROVED** [review]
but please do take a look at the comments

File: src/View/mainpage/IsodoseTab.py (hunk 1 above)
Comment (Fincap) [reply src/View/mainpage/IsodoseTab.py:40]: Whoops, I changed the type to a list while testing, and then changed it back to a dictionary but forgot to update the documentation. Will fix.

File: src/View/mainpage/IsodoseTab.py (hunk 2 above)
Comment (Fincap) [reply src/View/mainpage/IsodoseTab.py:80]: That's an interesting and powerful function, thanks for bringing it to my attention.

**Pull request merged**